packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -182,6 +162,112 @@ export class GitDiffWidget extends GitBaseWidget implements StatefulWidget {
             title: change.caption,
             className: 'status staged ' + GitFileStatus[change.status].toLowerCase()
         }, this.getStatusCaption(change.status, true).charAt(0)));
-        return h.div({ className: 'gitItem noselect' }, ...elements);
+        return h.div({ className: `gitItem noselect${change.selected ? ' ' + SELECTED_CLASS : ''}` }, ...elements);
+    }
+
+    protected handleRight(): void {
+        const selected = this.getSelected();
+        if (selected && GitFileChangeNode.is(selected)) {
+            const uri = this.getUriToOpen(selected);
+            this.editorManager.getByUri(uri).then(widget => {
+                if (widget instanceof EditorWidget && widget.editor instanceof MonacoDiffEditor) {
```

Comment:
actually there is no need to do the `instanceof` check. you'll get a non-navigable navigator for the code editor, see `DiffNavigator.canNavigate(): boolean`